AstarteDeviceSDKCSharp/Transport/MQTT/AstarteMqttV1Transport.cs

```diff
@@ -69,11 +69,20 @@ public AstarteMqttV1Transport(MutualSSLAuthenticationMqttConnectionInfo connecti
 
             try
             {
-                await DoSendMqttMessage(topic, payload, qos);
+                if (_client.TryPingAsync().Result)
```

Comment:
What happens here if a disconnection occurs immediately after the ping method or during `DoSendMqttMessage` method? I don't believe this approach solves all the issue related to duplicate and lost messages.

Reply:
If happens that method `DoSendMqttMessages` throw `MqtttCommunicationException`, exeption handling was here https://github.com/osmanhadzic/astarte-device-sdk-csharp-public/blob/ea7eb4e6ca83ce49f72c43f5ad4bc4f708be0488/AstarteDeviceSDKCSharp/Transport/MQTT/AstarteMqttV1Transport.cs#L197